src/p2pool.cpp

```diff
@@ -659,6 +659,36 @@ static bool init_signals()
 	return true;
 }
 
+void p2pool::stop()
+{
+	uv_loop_t *loop;
+
+	/* use async handle to make sure event loops wake up and stop */
+	{
+		uv_async_t asy;
+		loop = uv_default_loop();
+		uv_async_init(loop, &asy, NULL);
+		uv_stop(loop);
+		uv_async_send(&asy);
+	}
+
+	if (m_stratumServer) {
```

Comment:
I think it's better to move the code that stops Startum/P2P loops to their respective destructors.

Reply:
After more thinking, the best place for this async send would be shutdown_tcp() function. This will also avoid code duplication.

Reply:
Done